Cognite.Simulator.Tests/UtilsTests/ModelLibraryExtDepsUnitTest.cs

```diff
@@ -0,0 +1,191 @@
+
+            // Basic validations
+            Assert.NotNull(modelInState);
+            Assert.Equal("TestModelExternalId-v1", modelInState.ExternalId);
+            Assert.Equal(1, modelInState.DownloadAttempts);
+            Assert.True(modelInState.ParsingInfo.Parsed);
+            Assert.EndsWith(Path.Combine("100", "100.csv"), modelInState.FilePath);
+            Assert.Equal("csv", modelInState.FileExtension);
+            Assert.True(modelInState.Downloaded);
+
+            File.Exists(Path.Combine("100", "100.csv"));
+            File.Exists(Path.Combine("101", "101.xml"));
+            File.Exists(Path.Combine("102", "102.xml"));
+
+            Assert.NotNull(modelInState.DependencyFiles);
+            Assert.Equal(2, modelInState.DependencyFiles.Count);
+
+            var expectedDependencyFiles = Enumerable.Range(1, 2)
+                .Select(i => new DependencyFile
+                {
+                    Id = 100 + i,
+                    FilePath = Path.GetFullPath(Path.Combine($"{modelLibraryConfig.FilesDirectory}/{100 + i}", $"{100 + i}.xml")),
+                    Arguments = new Dictionary<string, string> { { "address", "test.address." + i } }
+                })
+                .ToArray();
+            Assert.Equivalent(expectedDependencyFiles, modelInState.DependencyFiles);
+
+            foreach (var mocker in endpointMockTemplates)
+            {
+                mocker.AssertCallCount();
+            }
+
+            VerifyLog(mockedLogger, LogLevel.Debug, "Model revision not found locally, adding to the local state: TestModelExternalId-v1", Times.Exactly(1), true);
+            VerifyLog(mockedLogger, LogLevel.Information, "Downloading 3 file(s) for model revision external ID: TestModelExternalId-v1", Times.Exactly(1), true);
+            VerifyLog(mockedLogger, LogLevel.Information, "Downloading file (1/3): 100. Model revision external ID: TestModelExternalId-v1", Times.Exactly(1), true);
+            VerifyLog(mockedLogger, LogLevel.Information, "Downloading file (2/3): 101. Model revision external ID: TestModelExternalId-v1", Times.Exactly(1), true);
+            VerifyLog(mockedLogger, LogLevel.Information, "Downloading file (3/3): 102. Model revision external ID: TestModelExternalId-v1", Times.Exactly(1), true);
+            VerifyLog(mockedLogger, LogLevel.Debug, "File downloaded: 100. Model revision: TestModelExternalId-v1", Times.Exactly(1), true);
+
+            // clear the state, run Init to load from the store to check the deserialization from LiteDB
+            modelInState.DependencyFiles.Clear();
+            Assert.NotNull(modelInState);
+            Assert.Empty(modelInState.DependencyFiles);
+            await lib.Init(CancellationToken.None); // TODO: we should be able to lib._state.Clear() & lib.Init() to restore state completely from LiteDB 
+            // And make sure redundant file download/processing is not happening https://cognitedata.atlassian.net/browse/POFSP-1138
+
```

Comment:
should we also add a test for what happens if one of the dependencies fails to download?